### PR TITLE
My list purchased albums

### PR DESCRIPTION
**Pull request by jxczz2**
## Summary

+  Functionality  list purchased albums is added
+  Acess validation is added
+ Tests for  list purchased albums is added

## Known Issues

no problem.

## Trello Card

https://trello.com/c/mkRyXhaN

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -11,11 +11,16 @@ exports.init = app => {
   app.post('/users', validatorMiddleware(signUpCheckValidations), userController.userCreate);
   app.post('/users/sessions', validatorMiddleware(signInCheckValidations), userController.signIn);
   app.get('/users', auth.verifyToken, userController.userList);
+  app.get(
+    '/users/:user_id/albums',
```
Comment (picollomartin) [inline app/routes.js]: Use camelCase for path params too (user_id)

File: app/middlewares/auth.js [hunk 2]
```diff
@@ -36,3 +36,12 @@ exports.verifyPermission = async (req, res, next) => {
     ? next(error.authorizationError('Is not  an Administrator user'))
     : next();
 };
+
+exports.verifyAccess = async (req, res, next) => {
+  const auth = req.headers[AUTHORIZATION];
+
+  const user = await getUser(auth);
```
Comment (picollomartin) [inline app/middlewares/auth.js]: You already obtain user in `verifyToken` middleware, put in some place like `req.user` and use here

File: test/album.js [hunk 3]
```diff
@@ -214,4 +215,119 @@ describe('albums', () => {
...
+                  expect(err.response.body).have.property('internal_code');
+                  expect(err.response.body.internal_code).to.equal('authorization_error');
+                  done();
+                });
+            });
+        });
+      });
+    });
+
+    it('should list all purchased albums from other user because is an admin user', done => {
```
Comment (picollomartin) [inline test/album.js:291]: Missing case of admin and his own albums (i know that is trivial)

File: test/album.js [hunk 4]
```diff
@@ -214,4 +215,119 @@ describe('albums', () => {
...
+          email: 'sarahi12hg@wolox.com',
+          password: 'woloxwoloA1520'
+        }).then(res => {
+          chai
+            .request(server)
+            .post('/albums/1')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .send()
+            .then(() => {
+              const admin = new User({
```
Comment (picollomartin) [inline test/album.js]: This code is repeated elsewhere, make a function for create admin users

File: test/album.js [hunk 5]
```diff
@@ -214,4 +215,119 @@ describe('albums', () => {
...
+                }).then(response => {
+                  chai
+                    .request(server)
+                    .get(`/users/1/albums`)
+                    .set(
+                      config.common.session.header_name,
+                      response.headers[config.common.session.header_name]
+                    )
+                    .then(result => {
+                      expect(result).have.status(200);
```
Comment (picollomartin) [inline test/album.js:327]: check response body

File: test/album.js [hunk 6]
```diff
@@ -214,4 +215,119 @@ describe('albums', () => {
...
+            .post('/albums/1')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .send()
+            .then(() => {
+              chai
+                .request(server)
+                .get(`/users/1/albums`)
+                .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+                .then(result => {
+                  expect(result).have.status(200);
```
Comment (picollomartin) [inline test/album.js:247]: check response body

**Review by picollomartin — CHANGES_REQUESTED** [review]
Nice work 😎

File: app/controllers/albums.js [hunk 7]
```diff
@@ -36,3 +36,12 @@ exports.buyAlbum = async (req, res, next) => {
     next(err);
   }
 };
+
+exports.listPurchasedAlbums = (req, res, next) =>
+  Album.getAlbumBy({
+    userId: req.params.userId
+  })
+    .then(albumUser => {
+      res.status(200).send(albumUser);
```
Comment (picollomartin) [inline app/controllers/albums.js]: Something very important is missing, you should call API for get the Data of every album

**Review by picollomartin — CHANGES_REQUESTED** [review]

File: app/controllers/albums.js [hunk 8]
```diff
@@ -36,3 +36,21 @@ exports.buyAlbum = async (req, res, next) => {
...
   }
 };
+
+exports.listPurchasedAlbums = async (req, res, next) => {
+  try {
+    const albumUser = await Album.getAllAlbumBy({
+      userId: req.params.userId
+    });
+
+    const promises = albumUser.map(async x => {
```
Comment (picollomartin) [inline app/controllers/albums.js]: Check declaration of things, `promises`- `x` are a little bit tricky of read and understand

File: app/controllers/albums.js [hunk 9]
```diff
@@ -36,3 +36,21 @@ exports.buyAlbum = async (req, res, next) => {
...
+      const album = await getResources(`/albums/${x.albumId}`);
```
Comment (picollomartin) [inline app/controllers/albums.js]: This await is unnecessary, do `albumUser.map(album => getResources(....))`

**Review by picollomartin — CHANGES_REQUESTED** [review]
Check declaration of things, names like `x` for variables is not quite good. Also, only as a comment, normally in test you call only the endpoint tested and all the data necessary is generated directly in the DB (with factory-girl or something like that)

File: app/controllers/albums.js (hunk 8 above)
Comment (jxczz2) [reply app/controllers/albums.js]: is correct, by mistake I corrected it in another branch

**Review by picollomartin — APPROVED** [review]

**Pull request merged**